cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -93,10 +94,14 @@
   public static final String CFG_HBASE_TABLE_COMPRESSION = "hbase.table.compression.default";
 
 
-  protected CConfiguration cConf;
+  protected String tablePrefix;
 
   public void setCConf(CConfiguration cConf) {
-    this.cConf = cConf;
+    this.tablePrefix = cConf.get(Constants.Dataset.TABLE_PREFIX);
+  }
+
+  protected boolean isCDAPTable(String hTableName) {
```

Comment:
Regardless of version, the full table name either begins with cdap. or cdap_
(no point in copying the same implementation to version specific methods).